cmd/soroban-cli/src/commands/contract/init.rs

```diff
@@ -1,85 +1,76 @@
-use clap::{
-    builder::{PossibleValue, PossibleValuesParser, ValueParser},
-    Parser,
-};
-use gix::{clone, create, open, progress, remote};
-use rust_embed::RustEmbed;
-use serde_json::{from_str, json, to_string_pretty, Error as JsonError, Value as JsonValue};
 use std::{
-    env,
-    ffi::OsStr,
-    fs::{
-        copy, create_dir_all, metadata, read_dir, read_to_string, write, File, Metadata,
-        OpenOptions,
-    },
-    io::{self, Read, Write},
-    num::NonZeroU32,
+    fs::{create_dir_all, write},
+    io,
     path::{Path, PathBuf},
     str,
-    sync::atomic::AtomicBool,
 };
-use toml_edit::{Document, TomlError};
-use ureq::get;
 
-use crate::{commands::global, print};
+use clap::Parser;
+use rust_embed::RustEmbed;
 
-const SOROBAN_EXAMPLES_URL: &str = "https://github.com/stellar/soroban-examples.git";
-const GITHUB_URL: &str = "https://github.com";
-const WITH_EXAMPLE_LONG_HELP_TEXT: &str =
-    "An optional flag to specify Soroban example contracts to include. A hello-world contract will be included by default.";
+use crate::commands::contract::init::Error::{
+    AlreadyExists, PathExistsNotCargoProject, PathExistsNotDir,
+};
+use crate::{commands::global, print};
 
 #[derive(Parser, Debug, Clone)]
 #[group(skip)]
 pub struct Cmd {
     pub project_path: String,
 
-    #[arg(short, long, num_args = 1.., value_parser=possible_example_values(), long_help=WITH_EXAMPLE_LONG_HELP_TEXT)]
-    pub with_example: Vec<String>,
+    #[arg(
+        long,
+        default_value = "hello-world",
+        long_help = "An optional flag to specify a new contract's name."
+    )]
+    pub name: String,
 
+    // TODO: remove in 23.0
     #[arg(
+        short,
         long,
-        default_value = "",
-        long_help = "An optional flag to pass in a url for a frontend template repository."
+        action = clap::ArgAction::HelpLong,
+        long_help = "This argument has been deprecated and will be removed in the future versions of CLI. You can still clone examples from the repo https://github.com/stellar/soroban-examples",
```

Comment:
Given that these deprecated switches are in fact noop, I think we should mention that the switch doesn't do anything anymore. The way is phrased right now gives the impression it worked, but won't in the future.

Reply:
If the switches don't do anything anymore, we should remove them so folks get a hard error if they try to use it.